### DISCUSSION TITLE
.NET 6.0 support

### DISCUSSION THREAD

**Discussion by JorenSynaeveTM** [dodona-edu/feedback-and-support#15] [Feature requests]
It looks like Dodona currently supports .NET Core 3.1 only. Is it possible to support .NET 6.0?
I think the judge needs to be updated for this.

We would like to start using Dodona, but our course will be updated to .NET 6.

**Comment by bmesuere** [comment]
The current C# judge was written by @dmouriss but as far as I know isn't used by him nor by any of us at the core team. This means that although it works, it isn't actively maintained.

We have already discussed the future of the judge internally. Our conclusion was that if there is demand for more support and/or features, we should add C# support to [TESTed](https://docs.dodona.be/nl/tested/) instead.

Can you explain in a bit more detail what you are planning to do with it?

@niknetniko Can you make an estimate for how much work it would take to add .NET 6.0 support to TESTed?

**Reply by JorenSynaeveTM** [reply]
In our beginner programming course, we would like to test the basic logic of applications using .NET 6 console applications. This means basic input - output tests as seen in the examples in [this repo](https://github.com/dodona-edu/example-exercises/tree/master/c%23). I have managed to get this working for our examples (currently in .NET Core 3.1), but some issues/bugs were discovered while testing. I have had contact with @dmouriss but, as you said, he is not actively maintaining the judge. 

Secondly, we would like to do some unit testing on classes in C#. Same issue here, I got it mostly working, but for .NET Core 3.1.

**Comment by dmouriss** [comment]
I am looking into updating the judge to .NET 6, but unfortunately this doesn't seem as straightforward as I hoped it would be.

**Comment by niknetniko** [comment]
We are actively working on adding support for C# (based on .NET 6) to TESTed: it should be ready in the near future.
